backend/gateway/src/handler/handler.service.ts

```diff
@@ -0,0 +1,57 @@
+import { User } from '@auth/dto/incoming/user.dto';
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { AxiosError } from 'axios';
+import { catchError, firstValueFrom } from 'rxjs';
+import { Employee } from './schemas/employee.schema';
+
+@Injectable()
+export class HandlerService {
+  private readonly EMPLOYEES_URL: string;
+
+  constructor(
+    private readonly httpService: HttpService
+  ) {
+    this.EMPLOYEES_URL = process.env.EMPLOYEES_URL;
+  }
+
+  async getEmployees(): Promise<Employee[]> {
+    const { data } = await firstValueFrom(
+      this.httpService
+        .get<Employee[]>(this.EMPLOYEES_URL + '/employees')
+        .pipe(
+          catchError((error: AxiosError) => {
+            throw 'An error happened! ' + error;
+          }),
+        ),
+    );
```

Comment:
this is fine for now but later I would implement some http class so you don't have to repeat the pipe, catch lines every time, just pass the url and it gives back the data

Reply:
I agree with that.